File: scripts/process-photos.mjs
import sharp from 'sharp';
import path from 'path';

const SRC = 'C:/Users/syste/OneDrive/Desktop/TwinOaksEngagement/Sparrow Photos/Professional Promotional Photos';
const DEST = 'C:/Users/syste/OneDrive/Desktop/TwinOaksEngagement/sparrow-website/public/images';

const tasks = [
  // Hero video poster — wide hillside panoramic, man overlooking Corvallis at dusk
  { src: 'SPR-6.jpg',                         dest: 'hero-poster.webp',           width: 1920 },
  // Hero image fallback — families walking together, warm community feel
  { src: 'SPR-11 (Hero Pic).jpg',             dest: 'hero-community-walk.webp',   width: 1920 },
  // Twin Oaks program card — man walking through the community interior
  { src: 'SPR-9.jpg',                         dest: 'twin-oaks/hero.webp',         width: 1200 },
  // LifeChange program card — families walking together
  { src: 'SPR-11 (Hero Pic).jpg',             dest: 'life-change/hero.webp',       width: 1200 },
  // Additional high-quality shots for future use
  { src: 'SPR-10.jpg',                        dest: 'twin-oaks-sign.webp',         width: 1200 },
  { src: 'SPR-16.jpg',                        dest: 'corvallis-golden-hour.webp',  width: 1920 },
  { src: 'SPR-18 (hero pic).jpg',             dest: 'hero-sunset.webp',            width: 1920 },
  { src: 'Twin_oaks-9169 (hero pic).jpg',     dest: 'twin-oaks-entrance.webp',     width: 900  },
  { src: 'Twin_oaks-9201.jpg',                dest: 'twin-oaks-home.webp',         width: 900  },
];

for (const { src, dest, width } of tasks) {
  const input  = path.join(SRC, src);
  const output = path.join(DEST, dest);
  await sharp(input)
    .resize({ width, withoutEnlargement: true })
    .webp({ quality: 82 })
    .toFile(output);
  const { size } = await import('fs').then(fs => fs.promises.stat(output));
  console.log(`✓  ${dest}  (${Math.round(size / 1024)} KB)`);
}

console.log('\nDone.');

File: scripts/process-photos.py
# /// script
# requires-python = ">=3.11"
# dependencies = ["Pillow"]
# ///
from pathlib import Path
from PIL import Image

SRC  = Path(r"C:\Users\syste\OneDrive\Desktop\TwinOaksEngagement\Sparrow Photos\Professional Promotional Photos")
DEST = Path(r"C:\Users\syste\OneDrive\Desktop\TwinOaksEngagement\sparrow-website\public\images")

tasks = [
    # Hero video poster — wide hilltop panoramic, man overlooking Corvallis at dusk
    ("SPR-6.jpg",                       "hero-poster.webp",          1920),
    # Hero image fallback — families walking, warm and human
    ("SPR-11 (Hero Pic).jpg",           "hero-community-walk.webp",  1920),
    # Twin Oaks program card — man walking through the community interior
    ("SPR-9.jpg",                       "twin-oaks/hero.webp",       1200),
    # LifeChange program card — families walking together
    ("SPR-11 (Hero Pic).jpg",           "life-change/hero.webp",     1200),
    # Additional shots for future use
    ("SPR-10.jpg",                      "twin-oaks-sign.webp",       1200),
    ("SPR-16.jpg",                      "corvallis-golden-hour.webp",1920),
    ("SPR-18 (hero pic).jpg",           "hero-sunset.webp",          1920),
    ("Twin_oaks-9169 (hero pic).jpg",   "twin-oaks-entrance.webp",    900),
    ("Twin_oaks-9201.jpg",              "twin-oaks-home.webp",        900),
]

for src_name, dest_rel, max_width in tasks:
    src_path  = SRC / src_name
    dest_path = DEST / dest_rel
    dest_path.parent.mkdir(parents=True, exist_ok=True)

    with Image.open(src_path) as img:
        # Preserve orientation from EXIF
        img = img.convert("RGB")
        w, h = img.size
        if w > max_width:
            new_h = int(h * max_width / w)
            img = img.resize((max_width, new_h), Image.LANCZOS)
        img.save(dest_path, "WEBP", quality=82, method=6)

    kb = dest_path.stat().st_size // 1024
    print(f"OK  {dest_rel}  ({kb} KB)")

print("\nDone.")
